src/main/kotlin/no/nav/klage/oppgave/eventlisteners/StatistikkTilDVHEventListener.kt

```diff
@@ -0,0 +1,112 @@
+package no.nav.klage.oppgave.eventlisteners
+
+import no.nav.klage.oppgave.domain.kafka.KlageStatistikkTilDVH
+import no.nav.klage.oppgave.domain.klage.*
+import no.nav.klage.oppgave.events.KlagebehandlingEndretEvent
+import no.nav.klage.oppgave.repositories.KlagebehandlingRepository
+import no.nav.klage.oppgave.repositories.MottakRepository
+import no.nav.klage.oppgave.service.StatistikkTilDVHKafkaProducer
+import no.nav.klage.oppgave.util.getLogger
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Service
+import java.time.LocalDateTime
+
+@Service
+class StatistikkTilDVHEventListener(
+    private val statistikkTilDVHKafkaProducer: StatistikkTilDVHKafkaProducer,
+    private val klagebehandlingRepository: KlagebehandlingRepository,
+    private val mottakRepository: MottakRepository
+) {
+
+    companion object {
+        @Suppress("JAVA_CLASS_ON_COMPANION")
+        private val logger = getLogger(javaClass.enclosingClass)
+    }
+
+    @EventListener
+    fun klagebehandlingEndretEventToDVH(klagebehandlingEndretEvent: KlagebehandlingEndretEvent) {
+        logger.debug("Received KlagebehandlingEndretEvent for klagebehandlingId ${klagebehandlingEndretEvent.klagebehandling.id} in StatistikkTilDVHEventListener")
+
+        val klagebehandling =
+            klagebehandlingRepository.findById(klagebehandlingEndretEvent.klagebehandling.id).orElseThrow()
+
+        val mottak = mottakRepository.getOne(klagebehandling.mottakId)
+
+        if (shouldSendStats(klagebehandlingEndretEvent.endringslogginnslag)) {
+            val klageStatistikkTilDVH = createDTO(
+                klagebehandling,
+                mottak,
+                getKlagebehandlingState(klagebehandlingEndretEvent.endringslogginnslag)
+            )
+            statistikkTilDVHKafkaProducer.sendStatistikkTilDVH(klageStatistikkTilDVH)
+        }
+    }
+
+    private fun shouldSendStats(endringslogginnslag: List<Endringslogginnslag>) =
+        endringslogginnslag.isEmpty() ||
+                endringslogginnslag.any { it.felt === Felt.TILDELT_SAKSBEHANDLERIDENT || it.felt === Felt.SLUTTFOERT }
+
+    private fun getKlagebehandlingState(endringslogginnslag: List<Endringslogginnslag>): String {
```

Comment:
Her ser jeg at du svarer på et spørsmål jeg stilte lenger opp.. :) Men burde dette vært enum-verdier, og ikke bare magiske strenger her midt i koden? Har vi bruk for disse statusene i andre sammenhenger mon tro? Burde det vært en del av domenet, og ikke bare blitt brukt i DVH-sammenheng? Eller er det fint å bare ha det her?

Reply:
Prøver du å bruke mine egne argument mot meg (om magiske strenger..)? ;) Jeg kan legge til et enum, men usikker på bruk utover DVH. We shall see..

Reply:
:-D